ui/pages/_app.js

```diff
@@ -363,39 +363,55 @@ class MesheryApp extends App {
   };
 
   loadOrg = async () => {
-    const { store } = this.props;
-    const currentOrgId = sessionStorage.getItem('currentOrgId');
+    const currentOrg = sessionStorage.getItem('currentOrg');
+    let reFetchKeys = false;
+
+    if (currentOrg && currentOrg !== 'undefined') {
+      let org = JSON.parse(currentOrg);
+      await this.loadAbility(org.id, reFetchKeys);
+      this.setOrganization(org);
+    }
+
     dataFetch(
       '/api/identity/orgs',
       {
         method: 'GET',
         credentials: 'include',
       },
-      (result) => {
+      async (result) => {
         let organizationToSet;
 
-        if (currentOrgId !== null) {
-          const indx = result.organizations.findIndex((org) => org.id === currentOrgId);
-          organizationToSet = indx !== -1 ? result.organizations[indx] : result.organizations[0];
+        if (currentOrg) {
+          const indx = result.organizations.findIndex((org) => org.id === currentOrg.id);
+          if (indx === -1) {
```

Comment:
Why are you only comparing to negative one? Aren’t there other possible values or scenarios in which the Org ID isn’t set but doesn’t equal negative one?

Reply:
No, if org doesn't match findIndex will only return -1 and that's when we refetch the keys otherwise we don't